### DISCUSSION TITLE
Avalonia WebAssembly and WebGL / Blazor / 3D

### DISCUSSION THREAD

**Discussion by Hecatron** [AvaloniaUI/Avalonia#10676] [General]
Hi,
So I've been looking through the past discussions around how to embed 3D into Avalonia.
One approach that might be possible that I wanted to ask about would be

  * Use [Photino](https://github.com/tryphotino/photino.NET) to host things at the top level
  * Have Avalonia in webassembly mode inside the Photino app
  * Then use the javascript interop to mount a webgl context somewhere in the form.
  * Use a 3D JS Library such as babylonjs to do the 3d work
  * Or if blazor could be mixed in with the Avalonia app use something like BabylonBlazor

So the question I've got would be is it possible to "mount" some other web content such as a blazor instance
within Avalonia in web assembly mode?

I did look at doing it the other way around of having a webview inside an Avalonia App
But there doesn't seem to be a consistent cross platform way of doing that at the moment

  * https://github.com/OutSystems/WebView - uses libCef instead of the windows webview2
  * https://github.com/maxkatz6/AvaloniaWebView
  * https://github.com/jpmikkers/Baksteen.Avalonia.Blazor

**Comment by maxkatz6** [comment]
It is possible to embed HTML content in the Avalonia app with NativeControlHost.
See https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog.Browser/EmbedSample.Browser.cs

**Comment by Hecatron** [comment]
thanks very much that's exactly what I was looking for
